### PR TITLE
snap/squashfs: fix minimum snap size on openSUSE

### PR DESCRIPTION
**Pull request by bboozzoo**
The mksquashfs build on some distros, eg. openSUSE, is patched to pad the filesystem image to 64k by default. Fix the unit tests to still pass on those distros, and leave a note that padding may not be as well controlled as thought.

Related change in snap/pack in  #13375

### CODE REVIEW

**Review by zyga — APPROVED** [review]

File: snap/squashfs/squashfs.go [hunk 1]
```diff
@@ -509,6 +509,8 @@ type BuildOpts struct {
 // partition table from the snap when a loopback device is created from it. If the snap
 // is smaller than this size, some versions of the kernel will print error logs while
 // scanning the loopback device for partitions.
+// TODO: this isn't reliable, on some distros mkfsquashfs pads up to 64k by
+// default
 const MinimumSnapSize int64 = 16384
```
Comment (pedronis) [inline snap/squashfs/squashfs.go:514]: does this mean we should increase this? or those distros make it bigger in mksquashfs but the kernel is happy with smaller anyway?

**Review by pedronis — COMMENTED** [review]
the test change is fine, but wondering if we need to do more?

File: snap/squashfs/squashfs.go (hunk 1 above)
Comment (bboozzoo) [reply snap/squashfs/squashfs.go:514]: No, AFAIU the default here is fine and correct for upstream squashfs-tools. OpenSUSE chose to patch squashfs-tools to pad to 64k https://build.opensuse.org/package/view_file/filesystems/squashfs/squashfs-64k.patch?expand=1 as a result of this bug https://bugzilla.suse.com/show_bug.cgi?id=953723 and they've been carrying the patch for 8 years now.

Comment (pedronis) [reply snap/squashfs/squashfs.go:514]: ok, then the TODO is a big confusing. maybe

// TODO: revisit if necessary, some distros patch user space to pad to 64k but kernel should work with this

**Review by pedronis — APPROVED** [review]
thank you, small suggestion for the TODO

**Pull request merged**